### PR TITLE
Do work in RPC call for getting file in its own thread

### PR DESCRIPTION
**Pull request by hmusum**

### CODE REVIEW

File: filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileDistributionRpcServer.java [hunk 1]
```diff
@@ -74,22 +79,24 @@ private void declareFileDistributionMethods() {
...
-        Optional<File> pathToFile = downloader.getFile(fileReference);
-        try {
-            if (pathToFile.isPresent()) {
-                req.returnValues().add(new StringValue(pathToFile.get().getAbsolutePath()));
-                log.log(LogLevel.DEBUG, "File reference '" + fileReference.value() + "' available at " + pathToFile.get());
-            } else {
-                log.log(LogLevel.INFO, "File reference '" + fileReference.value() + "' not found, returning error");
-                req.setError(fileReferenceDoesNotExists, "File reference '" + fileReference.value() + "' not found");
+        rpcDownloadExecutor.execute(() -> {
+            FileReference fileReference = new FileReference(req.parameters().get(0).asString());
```
Comment (baldersheim) [inline filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileDistributionRpcServer.java]: I prefer making it a private method and execute the private method. That gives more information on a stackdump. If not you have to consult code and to figure things out.

Comment (hmusum) [reply filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileDistributionRpcServer.java]: Good point, I had not thought about that, will do

**Comment by hmusum** [comment]
Update, please take a look again

File: filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileDistributionRpcServer.java [hunk 2]
```diff
@@ -74,22 +79,7 @@ private void declareFileDistributionMethods() {
...
-            } else {
-                log.log(LogLevel.INFO, "File reference '" + fileReference.value() + "' not found, returning error");
-                req.setError(fileReferenceDoesNotExists, "File reference '" + fileReference.value() + "' not found");
-            }
-        } catch (Throwable e) {
-            log.log(LogLevel.WARNING, "File reference '" + fileReference.value() + "' got exception: " + e.getMessage());
-            req.setError(fileReferenceInternalError, "File reference '" + fileReference.value() + "' removed");
-        }
-        req.returnRequest();
+        downloadFile(req);
```
Comment (baldersheim) [inline filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileDistributionRpcServer.java]: Do the execute here instead rpcDownloadExecutor.execute(() -> downloadFile())

File: filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileDistributionRpcServer.java [hunk 3]
```diff
@@ -123,4 +113,25 @@ public final void setFileReferencesToDownload(Request req) {
         req.returnValues().add(new Int32Value(0));
     }
 
+    private void downloadFile(Request req) {
+        rpcDownloadExecutor.execute(() -> {
```
Comment (baldersheim) [inline filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileDistributionRpcServer.java]: See above comment.

**Review by baldersheim — APPROVED** [review]

**Pull request merged**